paimon-spark/paimon-spark-common/src/main/scala/org/apache/paimon/spark/PaimonAnalysis.scala

```diff
@@ -28,6 +32,21 @@ class PaimonAnalysis(session: SparkSession) extends Rule[LogicalPlan] {
     case func: PaimonTableValueFunction if func.args.forall(_.resolved) =>
       PaimonTableValuedFunctions.resolvePaimonTableValuedFunction(session, func)
 
+    case o @ OverwritePartitionsDynamicPaimon(r, d) if o.resolved =>
+      PaimonDynamicPartitionOverwriteCommand(r, d, o.query, o.writeOptions, o.isByName)
   }
+}
 
+object OverwritePartitionsDynamicPaimon {
+  def unapply(o: OverwritePartitionsDynamic): Option[(DataSourceV2Relation, FileStoreTable)] = {
+    if (o.query.resolved) {
+      o.table match {
+        case r: DataSourceV2Relation if r.table.isInstanceOf[SparkTable] =>
```

Comment:
case r @ DataSourceV2Relation(r: SparkTable, _, _, ...) =>

Reply:
@YannByron The constructor of `DataSourceV2Relation` may change, the current code is more conducive to multi-version compatibility